File: code/C/ConsoleApplication1/ConsoleApplication1/bruteForce.c
#include <stdio.h>
#include <string.h> 
#include <math.h>

//int getSize(double *array) {
//    return (int)(sizeof(array) / sizeof(array[0]));
//}

//double bruteForce(double imageGS[752 * 752]) {
//    const double d = 3e-4;
//    const double t = 100e-6;
//    const double h = 7e-4;
//
//    printf("d = %lf\nt = %lf\nh = %lf\n", d, t, h);
//
//    const double height = 4.0e-3;
//    const double width = 4.0e-3;
//
//    const int pxW = 752;
//    const int pxH = 752;
//
//    const double pxSize = width / pxW;
//
//    double angle[360];
//
//    for (int i = 0; i < getSize(angle); i++) {
//        angle[i] = deg2rad(i);
//    }
//
//    long double x[360];
//    long double y[360];
//    memset(x, 0, 360); // заполнение массивов 0
//    memset(y, 0, 360); //
//
//    const int pxH = 1000;
//    const int pxW = 1000;
//    const pxSize = W / pxW;
//
//    const double theta_src = 65; // исходный зенитный угол
//    const double phi_src = 0;        // исходный азимутальный угол
//
//    const double theta = deg2rad(90) - deg2rad(theta_src);
//    const double phi = deg2rad(phi_src);
//    const double dC = h * tan(PI / 2 - theta);
//
//    double bruteArray = [bruteX, bruteY];
//    
//    return bruteArray;
//}

File: code/bruteForce.m
function [bruteX, bruteY] = bruteForce(imageGS)
    pxH = length(imageGS);
    pxW = length(imageGS(1, :));

    columns = pxH;
    rows = pxW;
    
    coloredX = [];
    coloredY = [];
    
    isFlag = false;
    isFlag2 = false;
    
    for  y = 1 : rows
        columnFlag = false;
        for x = 1 : columns
            if(imageGS(y, x) ~= 0)      
                coloredX(end + 1) = x;
                coloredY(end + 1) = y;
                
                columnFlag = true;
                isFlag2 = true;
                
                if (length(coloredX) == 1) 
                   isFlag = true; 
                end
            else
                if (columnFlag)
                   break; 
                end
            end
        end
        if (isFlag)
            if (~isFlag2)
               break; 
            end
        end
        isFlag2 = false;

    end

    if (length(coloredX) == 0 || length(coloredY) == 0)
       bruteX = 0;
       bruteY = 0;
       return;
    end
    
    bruteX = ceil(sum(coloredX) / length(coloredX));
    bruteY = ceil(sum(coloredY) / length(coloredY));
end